View post version history with differences

petersalomonsen opened the pull request:
**NOTE**: The preview URL below now contains the deployment of https://github.com/near/neardevhub-widgets/pull/197
I will change back to the contents of this PR once that other is completed.

Preview URL: https://near.org/devgovgigs.petersalomonsen.near/widget/gigs-board.pages.Post?id=8&timestamp=1688009948396559326&compareTimestamp=1688009918129563930

<img width="1000" alt="image" src="https://github.com/near/neardevhub-widgets/assets/9760441/43b35ccd-e534-48ad-87df-40666d68a042">

<img width="999" alt="image" src="https://github.com/near/neardevhub-widgets/assets/9760441/0882d555-a73b-4498-b988-23c2e79801dd">

Based on the work here: https://near.social/#/markeljan.near/widget/PostWithHistory?id=268&timestamp=1682382435187494750&compareTimestamp=1677857800212765894&referral=

resolves #64

ailisp commented:
I can't trigger the post history from your preview by clicking on the clock icon:

<img width="1360" alt="image" src="https://github.com/near/neardevhub-widgets/assets/13259400/f48db5e2-8372-4475-9776-59d05729c1fc">

Did you override the preview with the other branch?

petersalomonsen commented:
yes I did override the other branch. I've rebased on top of main now, and the preview url is working now, but I see now that there's an issue with post editing preview after rebase, so I need to look into that..

I've changed back to draft, and let you know as soon as I've fixed it.

petersalomonsen commented:
Issue with draft previews and history should be fixed now. Also added some more spacing on post confirmation as mentioned here ( https://github.com/near/neardevhub-widgets/pull/197#pullrequestreview-1520995410 )

ailisp commented on src/gigs-board/entity/post/Post.jsx:
```diff
@@ -651,16 +672,95 @@ const descriptionArea = isUnderPost ? (
+                swapTimestamps ? compareSnapshot : snapshot
+              ),
+              prevCode: combineText(
+                swapTimestamps ? snapshot : compareSnapshot
```
Nice normalization!

ailisp reviewed (approved):
Looks great! Thank you
